src/reducers/select.js

```diff
@@ -0,0 +1,197 @@
+/**
+*
+* @licstart  The following is the entire license notice for the JavaScript code in this file.
+*
+* Merge MARC records
+*
+* Copyright (C) 2015-2019 University Of Helsinki (The National Library Of Finland)
+*
+* This file is part of marc-record-merge-js
+
+* marc-record-merge-js program is free software: you can redistribute it and/or modify
+* it under the terms of the GNU Lesser General Public License as
+* published by the Free Software Foundation, either version 3 of the
+* License, or (at your option) any later version.
+*
+* marc-record-merge-js is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+* GNU Lesser General Public License for more details.
+*
+* You should have received a copy of the GNU Lesser General Public License
+* along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*
+* @licend  The above is the entire license notice
+* for the JavaScript code in this file.
+*
+*/
+import createDebugLogger from 'debug';
+import {normalizeSync} from 'normalize-diacritics';
+
+export default (pattern) => (base, source) => {
+  const debug = createDebugLogger('@natlibfi/marc-record-merge');
+  const baseFieldsFromPattern = base.get(pattern);
+  const sourceFieldsFromPattern = source.get(pattern);
+  return selectFields();
+
+  function selectFields() {
+    // Check that all fields are data fields
+    // Test 01: The field is a control field (contains 'value')
+    const baseFields = checkFieldType(baseFieldsFromPattern);
+    //debug(`baseFields: ${JSON.stringify(baseFields, undefined, 2)}`);
+    const sourceFields = checkFieldType(sourceFieldsFromPattern);
+    //debug(`sourceFields: ${JSON.stringify(sourceFields, undefined, 2)}`);
+      
+    // Test 02: If there are multiple fields, the base field is returned unmodified
+    if (baseFields.length > 1 || sourceFields.length > 1) {
+      debug(`Multiple fields in base or source`);
+      return base;
+    }
+
```

Comment:
You can now assign the individual fields to variables because there's no need to access them in arrays. So, like so:
```js
const [baseField] = baseFields;
const [sourceField] = sourceFields;
```